tests/std/tests/P2408R5_ranges_iterators_to_classic_algorithms/test.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <execution>
+#include <ranges>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+template <class T, class U>
+using second = U;
+template <class, class Input>
+Input second_v(Input x) {
+    return x;
+}
+
+template <class I>
+struct iterator_adaptor {
+    std::vector<int> v;
+    iterator_adaptor(initializer_list<int> il) : v(il) {}
+    iterator_adaptor(size_t n) : v(n) {}
+
+    using iterator       = I;
+    using const_iterator = typename I::Consterator;
+
+    iterator begin() {
+        return iterator{v.data()};
+    }
+    iterator end() {
+        return iterator{v.data() + v.size()};
+    }
+
+    const_iterator cbegin() const {
+        return const_iterator{v.data()};
+    }
+    const_iterator cend() const {
+        return const_iterator{v.data() + v.size()};
+    }
+};
+
+template <class... Is>
+struct helper {
+    helper(second<Is, initializer_list<int>>... ils) : tup(ils...) {}
+
+    helper(initializer_list<int> il) requires(sizeof...(Is) > 1) : tup(second_v<Is>(il.size())...) {
+        get<0>(tup).v.assign(il);
+    }
+
+    std::tuple<iterator_adaptor<Is>...> tup;
+};
+
+template <size_t Idx, class... Is>
+auto hbegin(helper<Is...>& h) {
+    return get<Idx>(h.tup).begin();
+}
+template <size_t Idx, class... Is>
+auto hend(helper<Is...>& h) {
+    return get<Idx>(h.tup).end();
+}
+
+template <size_t Idx, class... Is>
+auto hcbegin(const helper<Is...>& h) {
+    return get<Idx>(h.tup).cbegin();
+}
+template <size_t Idx, class... Is>
+auto hcend(const helper<Is...>& h) {
+    return get<Idx>(h.tup).cend();
+}
+
+template <class It>
+struct unary_algorithms {
+    static void call() {
+        if constexpr (forward_iterator<It>) {
+            // parallel algorithms
+            using execution::seq;
+            {
+                helper<It> h({1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+                auto res = reduce(seq, hcbegin<0>(h), hcend<0>(h), 0);
+                assert(res == 55);
+            }
+        }
+    }
+};
+
+template <class I1, class I2>
+struct binary_algorithms {
+    static void call() {
+        if constexpr (forward_iterator<I1> || _Is_fwd_iter_v<I2>) {
+            helper<I1, I2> h({0, 0, 1, 2, 3, 3, 4, 5});
+            initializer_list exp{0, 1, 2, 3, 4, 5};

```

Comment:
This line and similar lines failed due to LLVM-40378... We need to write `initializer_list<int>` to support Clang.